algoliasearch-core/src/main/java/com/algolia/search/models/rules/FacetValuesOrder.java

```diff
@@ -0,0 +1,63 @@
+package com.algolia.search.models.rules;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import java.io.Serializable;
+import java.util.List;
+import java.util.Objects;
+
+/** Facet values ordering rule container. */
+@JsonInclude(JsonInclude.Include.NON_NULL)
+public class FacetValuesOrder implements Serializable {
+  private List<String> order;
+  private String sortRemainingBy;
+
+  public FacetValuesOrder() {}
+
+  public FacetValuesOrder(List<String> order, String sortRemainingBy) {
+    this.order = order;
+    this.sortRemainingBy = sortRemainingBy;
+  }
+
+  public List<String> getOrder() {
+    return order;
+  }
+
+  public FacetValuesOrder setOrder(List<String> order) {
+    this.order = order;
+    return this;
+  }
+
+  public String getSortRemainingBy() {
```

Comment:
can this be more restricted than string?

Reply:
We can, but this client's philosophy avoids introducing restrictions (this is observable in many other places in the codebase)